### PR TITLE
Fix an issue where null values were causing an error in automation loops

### PR DESCRIPTION
**Pull request by PClmnt**
## Description
Null or undefined values were causing an issue in the loop binding replacement code and throwing an error. This fix simply does nothing for those values, (since they're empty and wont contain and handlebars statements that need updated). 

Also added some tests to cover this issue and potential other problem areas such as empty arrays / objects. 
## Addresses
https://linear.app/budibase/issue/BUDI-8410/looping-in-automations-cannot-convert-undefined-or-null-to-object

## Launchcontrol
- Fixes an issue where looping an object or array with null values was causing undefined errors

### CODE REVIEW

File: packages/server/src/automations/loopUtils.ts [hunk 1]
```diff
@@ -18,14 +18,18 @@ function replaceBindingsRecursive(
...
+
   if (typeof value === "object") {
     for (const [innerKey, innerValue] of Object.entries(value)) {
       if (typeof innerValue === "string") {
         value[innerKey] = automationUtils.substituteLoopStep(
           innerValue,
           `steps.${loopStepNumber}`
         )
-      } else if (typeof innerValue === "object") {
+      } else if (typeof innerValue === "object" && innerValue !== null) {
```
Comment (samwho) [inline packages/server/src/automations/loopUtils.ts:32]: I know this isn't your code but `typeof innerValue === "object"` almost certainly doesn't do what the author is expecting it to. 

```
❯ node
Welcome to Node.js v20.10.0.
Type ".help" for more information.
> typeof {} === "object"
true
> typeof [] === "object"
true
> typeof null === "object"
true
```

**Review by samwho — APPROVED** [review]
Beautiful tests.

File: packages/server/src/automations/loopUtils.ts (hunk 1 above)
Comment (PClmnt) [reply packages/server/src/automations/loopUtils.ts:32]: I think it is mine from a good while ago - but yes you're right. We would be better checking for the existence of keys in the object here I think

Comment (PClmnt) [reply packages/server/src/automations/loopUtils.ts]: UPD2: I added some more checks in that if, in order to be a bit more clear on what we're doing.

File: packages/server/src/automations/loopUtils.ts [hunk 2]
```diff
@@ -29,7 +29,12 @@ function replaceBindingsRecursive(
           innerValue,
           `steps.${loopStepNumber}`
         )
-      } else if (typeof innerValue === "object" && innerValue !== null) {
+      } else if (
+        innerValue &&
+        typeof innerValue === "object" &&
+        !Array.isArray(innerValue) &&
+        Object.keys(innerValue).length > 0
```
Comment (samwho) [inline packages/server/src/automations/loopUtils.ts:36]: If you're wanting to detect that something is an object, `_.isPlainObject` is a bit clearer.

Comment (mike12345567) [reply packages/server/src/automations/loopUtils.ts:36]: this can be re-written to
```
import { isPlainObject } from "lodash"
....
innerValue && isPlainObject(innerValue) && Object.keys(innerValue).length > 0
```

Comment (PClmnt) [reply packages/server/src/automations/loopUtils.ts]: Done, thanks!
